### PR TITLE
- Added Update-VSTeamGitRepositoryDefaultBranch to allow for changing…

### PR DESCRIPTION
**Pull request by rbleattler**
# PR Summary

- Added Update-VSTeamGitRepositoryDefaultBranch to allow for changing the default branch of a repository.
- Fixed bad link in README.md

## PR Checklist

- [X] [Write Help](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#write-help)
- [X] [Write Unit Test](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#write-unit-test)
- [X] [Update CHANGELOG.md](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#update-changelogmd)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+- Added Update-VSTeamGitRepositoryDefaultBranch to allow for changing the default branch of a repository.
```
Comment (SebastianSchuetze) [inline CHANGELOG.md]:
```suggestion
## 7.8.0

- Added Update-VSTeamGitRepositoryDefaultBranch to allow for changing the default branch of a repository.
```

Comment (SebastianSchuetze) [reply CHANGELOG.md]: For a new freature you can raise the minor version. Can you please do the same in the PS manifest?

File: Source/Public/Update-VSTeamGitRepositoryDefaultBranch.ps1 [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+         $resp = _callAPI -Method PATCH -ProjectName $ProjectName `
+            -Area "git" `
+            -Resource "repositories" `
+            -id $Repo.Id `
+            -Body $body `
+            -Version $(_getApiVersion Git)
+         # Storing the object before you return it cleaned up the pipeline.
+         # When I just write the object from the constructor each property
+         # seemed to be written
+         # $repo = [vsteam_lib.GitRepository]::new($resp, $ProjectName)
```
Comment (SebastianSchuetze) [inline Source/Public/Update-VSTeamGitRepositoryDefaultBranch.ps1]: Could you explain this to me?

File: Tests/function/tests/Update-VSTeamGitRepositoryDefaultBranch.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      #       Update-VSTeamGitRepositoryDefaultBranch -Name PeopleTracker -projectname PeopleTracker -DefaultBranch 'develop'
+
+      #       ## Assert
+      #       Should -Invoke Invoke-RestMethod -ParameterFilter {
+      #          Write-Debug "Method : $Method"
+      #          Write-Debug "Uri : $Uri"
+      #          $Method -eq 'Patch' -and $Uri -eq "https://localhost:8080/tfs/PeopleTracker/PeopleTracker/_apis/git/repositories/00000000-0000-0000-0000-000000000000?api-version=$(_getApiVersion Git)"
+      #       }
+      #    }
+      # }
```
Comment (SebastianSchuetze) [inline Tests/function/tests/Update-VSTeamGitRepositoryDefaultBranch.Tests.ps1]: Please remove it if not needed. Or are you not finished yet?

Or do you try to solve something specific?

File: Tests/function/tests/Update-VSTeamGitRepositoryDefaultBranch.Tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            Mock _getInstance { return 'https://dev.azure.com/Test' }
+         }
+
+         It "by name should update Git repo's default branch" {
+            ## Act
+            Update-VSTeamGitRepositoryDefaultBranch -Name PeopleTracker -projectname PeopleTracker -DefaultBranch 'develop'
+
+            ## Assert
+            Should -Invoke Invoke-RestMethod -ParameterFilter {
+               Write-Debug "Method : $Method"
```
Comment (SebastianSchuetze) [inline Tests/function/tests/Update-VSTeamGitRepositoryDefaultBranch.Tests.ps1]: Mhh this works? Never thought of that!

File: Source/Public/Update-VSTeamGitRepositoryDefaultBranch.ps1 [hunk 5]
```diff
@@ -0,0 +1,61 @@
+# Sets the default branch for a Git repository in your Azure DevOps or Team Foundation Server account.
+#
+# Get-VSTeamOption 'git' 'repositories'
+# id              : 225f7195-f9c7-4d14-ab28-a83f7ff77e1f
+# area            : git
+# resourceName    : repositories
+# routeTemplate   : {project}/_apis/{area}/{resource}/{repositoryId}
+# http://bit.ly/Add-VSTeamGitRepository
```
Comment (SebastianSchuetze) [inline Source/Public/Update-VSTeamGitRepositoryDefaultBranch.ps1]:
```suggestion
# http://bit.ly/Add-VSTeamGitRepository
```
Please update this to the same name as the cmdlet

File: Source/Public/Update-VSTeamGitRepositoryDefaultBranch.ps1 [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+      [Parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true)]
+      [vsteam_lib.ProjectValidateAttribute($false)]
+      [ArgumentCompleter([vsteam_lib.ProjectCompleter])]
+      [string] $ProjectName,
+      [Parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true)]
+      [string] $DefaultBranch
+   )
+   begin {
+      try {
+         # $Repo = Get-VSTeamGitRepository -Name $Name -ProjectName $ProjectName
```
Comment (SebastianSchuetze) [inline Source/Public/Update-VSTeamGitRepositoryDefaultBranch.ps1]: Why are you not using this cmdlet? We are fine to use existing cmdlet to make more complex ones.

File: Source/Public/Update-VSTeamGitRepositoryDefaultBranch.ps1 [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+      }
+      if ($DefaultBranch -notlike "*refs/head*") {
+         $DefaultBranch = 'refs/head/{0}' -f $DefaultBranch
+      }
+   }
+   process {
+      
+      
+      $body = @{  
+         # name          = $Name
```
Comment (SebastianSchuetze) [inline Source/Public/Update-VSTeamGitRepositoryDefaultBranch.ps1]: Also remove as soon as it is clear what you wanna do.

**Review by SebastianSchuetze — CHANGES_REQUESTED** [review]
Hey @rbleattler thank you for the contribution!

After reviewing it feels like you are not completely sure about your approach with some parts of the code. We need to clarify and clean this up before we can merge it!

File: Tests/function/tests/Update-VSTeamGitRepositoryDefaultBranch.Tests.ps1 (hunk 3 above)
Comment (rbleattler) [reply Tests/function/tests/Update-VSTeamGitRepositoryDefaultBranch.Tests.ps1]: I was trying to follow convention used in a different test but am not particularly proficient with pester and wanted to get this out there. Apologies for not removing it.

File: Tests/function/tests/Update-VSTeamGitRepositoryDefaultBranch.Tests.ps1 (hunk 4 above)
Comment (rbleattler) [reply Tests/function/tests/Update-VSTeamGitRepositoryDefaultBranch.Tests.ps1]: The debug lines (meant to remove, again sorry -- working on this _at work_ while working on some migration stuff so in a time crunch) or the assertion itself?

File: Source/Public/Update-VSTeamGitRepositoryDefaultBranch.ps1 (hunk 6 above)
Comment (rbleattler) [reply Source/Public/Update-VSTeamGitRepositoryDefaultBranch.ps1]: Honestly, I believe I intended to be revert this once the tests were working, but this is related to the aforementioned issue with Pester.

File: Source/Public/Update-VSTeamGitRepositoryDefaultBranch.ps1 (hunk 2 above)
Comment (rbleattler) [reply Source/Public/Update-VSTeamGitRepositoryDefaultBranch.ps1]: This was a remnant from the function I used as a basis for creating this. Will remove and re-commit.

File: Tests/function/tests/Update-VSTeamGitRepositoryDefaultBranch.Tests.ps1 (hunk 3 above)
Comment (rbleattler) [reply Tests/function/tests/Update-VSTeamGitRepositoryDefaultBranch.Tests.ps1]: Cleaning up, and re-committing.

File: Source/Public/Update-VSTeamGitRepositoryDefaultBranch.ps1 (hunk 6 above)
Comment (rbleattler) [reply Source/Public/Update-VSTeamGitRepositoryDefaultBranch.ps1]: Fixed, re-committing.

File: Source/Public/Update-VSTeamGitRepositoryDefaultBranch.ps1 (hunk 7 above)
Comment (rbleattler) [reply Source/Public/Update-VSTeamGitRepositoryDefaultBranch.ps1]: Cleaning up, and re-committing.

**Comment by rbleattler** [comment]
Closing and re-opening with proposed changes.

**Pull request closed without merging**